proto/workflow_execution_service.proto

```diff
@@ -0,0 +1,237 @@
+  // endpoint is that the workflow_params JSON will include parameterizations along with
+  // input and output files.  The latter two may be on S3, Google object storage, local filesystems,
+  // etc.  This specification makes no distinction.  However, it is assumed that the submitter
+  // is using URLs that this system both understands and can access. For Amazon S3, this could
+  // be accomplished by given the credentials associated with a WES service access to a
+  // particular bucket.  The details are important for a production system and user on-boarding
+  // but outside the scope of this spec.
+  rpc RunWorkflow(workflow_request) returns (workflow_run_id) {
+    option (google.api.http) = {
+      post: "/ga4gh/wes/v1/workflows"
+      body: "*"
+    };
+  }
+
+  // Get quick status info about a running workflow.
+  rpc GetWorkflowStatus(workflow_run_id) returns (workflow_status) {
+      option (google.api.http) = {
+        get: "/ga4gh/wes/v1/workflows/{workflow_id}/status"
+      };
+  }
+
+  // Get detailed info about a running workflow.
+  rpc GetWorkflowLog(workflow_run_id) returns (workflow_log) {
+      option (google.api.http) = {
+        get: "/ga4gh/wes/v1/workflows/{workflow_id}"
+      };
+  }
+
+  // Cancel a running workflow.
+  rpc CancelJob(workflow_run_id) returns (workflow_run_id) {
+    option (google.api.http) = {
+      delete: "/ga4gh/wes/v1/workflows/{workflow_id}"
+    };
+  }
+
+  // List the workflows, this endpoint will list the workflows in order of oldest to newest.
+  // There is no guarantee of live updates as the user traverses the pages, the behavior should be
+  // decided (and documented) by each implementation.
+  // To monitor a given execution, use GetWorkflowStatus or GetWorkflowLog.
+  rpc ListWorkflows(workflow_list_request) returns (workflow_list_response) {
+    option (google.api.http) = {
+      get: "/ga4gh/wes/v1/workflows"
+    };
+  }
+
+}
+
+// To execute a workflow, send a workflow request including all the details needed to begin downloading
+// and executing a given workflow.
+message workflow_request {
+  // OPTIONAL
+  // The workflow CWL or WDL document, must provide either this or workflow_url
+  string workflow_descriptor = 1;
+
+  // REQUIRED
+  // The workflow parameterization document (typically a JSON string), includes all parameterizations for
+  // the workflow including input and output file locations
+  string workflow_params = 2;
+
+  // REQUIRED
+  // The workflow descriptor type, must be "CWL" or "WDL" currently (or another alternative supported by this WES instance)
+  string workflow_type = 3;
+
+  // REQUIRED
+  // The workflow descriptor type version, must be one supported by this WES instance
+  string workflow_type_version = 4;
+
+  // OPTIONAL
+  // A key-value map of arbitrary metadata outside the scope of the workflow_params but useful to track with this workflow request
+  map<string,string> key_values = 5;
```

Comment:
Same concept as the TES `tags` ( https://github.com/ga4gh/task-execution-schemas/blob/master/task_execution.proto#L66 ) but with a different name